test/e2e/e2e.go

```diff
@@ -1942,17 +1942,17 @@ var _ = ginkgo.Describe("e2e ingress traffic validation", func() {
 						}
 
 						// After the DualStack upgrade, we might hit a timeout on the Service.
-						// Give this 5 tries to settle
+						// Give this up to 30 seconds to settle.
 						// This could be implemented way more elegantly with gomega 1.14.0 (dependency on k8s 1.22) and above, see:
 						// https://github.com/onsi/gomega/commit/2f04e6e3467d2c0c695786c3e7880f1e940274cf#
 						ginkgo.By(fmt.Sprintf("Waiting for the service to stabilize after the DualStack upgrade on address %s (node %s)", nodeAddress.Address, node.Name))
 						gomega.Eventually(func() (r bool) {
 							failures := gomega.InterceptGomegaFailures(func() {
 								framework.Logf("Trying to poke node %s with address %s and port %s/%d", node.Name, nodeAddress.Address, protocol, nodePort)
-								pokeEndpointHostname(clientContainerName, protocol, nodeAddress.Address, nodePort)
+								pokeEndpointHostnameWithTimeout(clientContainerName, protocol, nodeAddress.Address, nodePort, 4)
 							})
 							return len(failures) == 0
-						}, 5*time.Second, 1*time.Second).Should(gomega.BeTrue())
+						}, 30*time.Second, 5*time.Second).Should(gomega.BeTrue())
```

Comment:
the PR description says 120 seconds... not sure what that means...

- could you please explain how/why you picked 4 seconds max-timeout for the `pokeEndpointHostnameWithTimeout` -> is it random for a curl timeout value? By doing so the failures that will be caught here will have less flakes?
- so here are we bumping total timeout of e2e to 30seconds with each try having 5 second periodic interval for retry within those 30seconds?
I am trying to understand what's the main reason behind the flake, is it the total time out for the test being too less or is it that curl takes time for this test and sometimes will fail the first few times?

Reply:
oh wait, I guess its really 30seconds for each try and 5 tries so total max of 120seconds :)

Reply:
Actually no. I had opened this with 120 but then lowered it to 30 and then didn't update the PR description = my bad

Reply:
I wanted this to have some time to settle and just try each of the ports for x amount of time to make sure that everything was stable after the DualStack migration.

However, before, I gave it only 5 seconds max and polled at one second intervals but pokeEndpointHostname had no timeout at all meaning that in case of a failed TCP connection attempt, it would just sit there (I think the default curl timeout is like 120 or so) and it would just try once.
That specific flake https://github.com/ovn-org/ovn-kubernetes/runs/5634890808?check_suite_focus=true#step:15:155 fails exactly after that one attempt at the 5 second mark.

Now, I poll for a total of 30 seconds, at 5 second intervals (so 6/7 times total)  and I set the curl max-time to 4 seconds so that a single curl call will never take more than 4 seconds total (I haven't tried bringing it to 5, I just put it below the poll interval for "safety"). That way, stale TCP connections will be killed if they take too long and I simply retry:
https://github.com/ovn-org/ovn-kubernetes/runs/5642015403?check_suite_focus=true#step:15:312

I still don't like that code, though, let me go back to the drawing board; there's also a panic in https://github.com/ovn-org/ovn-kubernetes/runs/5642015403?check_suite_focus=true#step:15:107 which wasn't even reported by CI.